src/sql/sqlops.proposed.d.ts

```diff
@@ -325,4 +325,28 @@ declare module 'sqlops' {
 			}
 		}
 	}
+
+	/**
+	 * Namespace for interacting with query editor
+	*/
+	export namespace queryEditor {
+		/**
+		 * Create new query editor
+		 * @param {string} queryContent The content of query editor
+		 * @returns {string} The file URI of the new query editor
+		*/
+		export function newQueryEditor(queryContent?: string): Thenable<string>;
+
+		/**
+		 * Make connection for the query editor
+		 * @param {string} fileUri file URI for the query editor
+		 * @param {string} connectionId connection ID
+		 */
+		export function connect(fileUri: string, connectionId: string): Thenable<void>;
+
+		/**
+		 * Locates the active editor and run query if it is a query editor.
+		 */
+		export function runCurrentQuery(): void;
```

Comment:
This API would make more sense if it took the editor URI as an argument and executed the query on the given editor. Is there some reason not to do that?

Reply:
Fully agree on this. What might be useful is an overload method to get the active SQL editor. There's already code to get the active editor(s) given certain filters but given how common this will be, it merits a new function.

Reply:
What do you mean? The current implementation is geting the active SQL editor

---
In reply to: [183219093](https://github.com/Microsoft/sqlopsstudio/pull/1196#discussion_r183219093) [](ancestors = 183219093)